### ISSUE TITLE
How to predict the new image by using model.predict?

### ISSUE THREAD

**Issue by yashfujifilm** [keras-team/keras#6993]
I have built a model and saved its weights as 'first_try.h5' but I am not able to load the model & run it on any random image. My problem is how to use model.predict_generator or model.predict or model.predict_on_batch methods. I have read the documentation, but I need one small example to do so. Sorry, I am new to this. It is a binary classifier.

I am using CNTK as my backend. I am following this[ tutorial.](https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html)

Need little help.

EDIT: I am trying the following way, but the output is just [[0.]]

'model.load_weights('first_try.h5')
img = Image.open("data/predict/T160305M-0222c_5.jpg")
a = array(img).reshape(1,512,512,3)
score = model.predict(a, batch_size=32, verbose=0)
print(score)'

**Comment by tejareddy-b** [comment]
```
I had same problem.
I was not saving the weights properly but now everything seem fine.
Please add the save_weights code so that we can know where the problem is.
```

**Comment by blattnem** [comment]
Probably you have to normalise the numpy array by 255.

**Comment by punnerud** [comment]
1. When you load weight, you have to have something to load them into. So first you have to load the model, before your weight. Guide on how to save your model to JSON/YAML: https://machinelearningmastery.com/save-load-keras-deep-learning-models/
2. Load your weight (you got that)
3. Preprocess the image in the same way as you did with the training (everything have to be in range between 0 and 1. In your case img/255)
4. Now you can do the prediction. Remember to include a fictive ID as an extra dimension, because you most likely had that in your training, to separate the trainingdata. For a single image you can solve it this way:
```
import numpy as np
model.predict(np.expand_dims(img, axis=0))
```

Let me know if you need a complete example on training, saving model+weight and prediction.

**Comment by naren142** [comment]
@punnerud where can I get the complete example??

**Comment by ohmerhe** [comment]
I also encounter the preprocess problem,  i am not sure what shall i do in preprocess

**Comment by punnerud** [comment]
Sorry, I have moved most of my work to PyTorch now. Don’t think I still have the complete example in Keras.

**Comment by Rachquazar** [comment]
kllllllllllllllll

**Comment by aofek** [comment]
Thank you very much punnerud !!!!!!!
adding this dimension solved a problem which held me back for 10 hours
Thanks again
aviofek
